bgpd/bgp_route.c

```diff
@@ -10552,45 +10552,178 @@ DEFUN (show_ip_bgp_large_community,
 }
 
 static int bgp_table_stats(struct vty *vty, struct bgp *bgp, afi_t afi,
-			   safi_t safi);
+			   safi_t safi, struct json_object *json);
 
 
+DEFUN(show_ip_bgp_statistics_all, show_ip_bgp_statistics_all_cmd,
+      "show [ip] bgp [<view|vrf> VIEWVRFNAME] statistics-all [json]",
+      SHOW_STR IP_STR BGP_STR BGP_INSTANCE_HELP_STR
+      "Display number of prefixes for all afi/safi\n" JSON_STR)
+{
+	bool uj = use_json(argc, argv);
+	struct bgp *bgp = NULL;
+	safi_t safi;
+	afi_t afi;
+	int idx = 0;
+	struct json_object *json_all = NULL;
+	struct json_object *json_afi_safi = NULL;
+
+	bgp_vty_find_and_parse_afi_safi_bgp(vty, argv, argc, &idx, &afi, &safi,
+					    &bgp, false);
+	if (!bgp)
+		return CMD_WARNING;
+
+	if (uj)
+		json_all = json_object_new_object();
+
+	FOREACH_AFI_SAFI (afi, safi) {
+		/*
+		 * So limit output to those afi/safi pairs that
+		 * actually have something interesting in them
+		 */
+		if (strmatch(get_afi_safi_str(afi, safi, true),
+			     "Unknown")) {
+			continue;
+		}
+		if (uj)
+			json_afi_safi = json_object_new_array();
+		else
+			json_afi_safi = NULL;
+
+		bgp_table_stats(vty, bgp, afi, safi, json_afi_safi);
+
+		if (uj)
+			json_object_object_add(
+					       json_all,
```

Comment:
Did clang-format do this? Unless github formatting is deceiving me, I think this can go on the previous line